src/main/java/com/uid2/shared/middleware/AuthMiddleware.java

```diff
@@ -198,6 +214,24 @@ public void handle(RoutingContext rc) {
             }
         }
 
+        public void handleV2(RoutingContext rc, Role role) {
+            // add aws request id tracer to help validation
+            String traceId = rc.request().getHeader("X-Amzn-Trace-Id");
+            if (traceId != null && traceId.length() > 0) {
+                rc.response().headers().add("X-Amzn-Trace-Id", traceId);
+            }
+
+            final String authHeaderValue = rc.request().getHeader(AuthMiddleware.AuthorizationHeader);
+            final String authKey = AuthHandler.extractBearerToken(authHeaderValue);
+            final ClientKey profile = (ClientKey)this.authKeyStore.get(authKey);
+            AuthMiddleware.setAuthClient(rc, profile);
+            if (profile == null || profile.isDisabled() || !profile.getRoles().contains(role)) {
```

Comment:
What is the reason for not using the `authorizationProvider` here? How is this method different from the `handle() method above?

Reply:
Reverted. authorizationProvider seems a little overkill.